1.1.18/docker-entrypoint.sh

```diff
@@ -1,9 +1,72 @@
-#!/bin/sh
-set -e
+#!/bin/bash
+set -eo pipefail
+CONFIG_FILE='/etc/gearmand.conf'
+VERBOSE=${VERBOSE:-INFO}
+QUEUE_TYPE=${QUEUE_TYPE:-builtin}
+THREADS=${THREADS:-4}
+JOB_RETRIES=${JOB_RETRIES:-0}
+KEEPALIVE=${KEEPALIVE:-0}
+KEEPALIVE_IDLE=${KEEPALIVE_IDLE:-30}
+KEEPALIVE_INTERVAL=${KEEPALIVE_IDLE:-10}
+KEEPALIVE_COUNT=${KEEPALIVE_COUNT:-5}
+
+# usage: file_env VAR [DEFAULT]
+#    ie: file_env 'XYZ_DB_PASSWORD' 'example'
+# (will allow for "$XYZ_DB_PASSWORD_FILE" to fill in the value of
+#  "$XYZ_DB_PASSWORD" from a file, especially for Docker's secrets feature)
+file_env() {
+	local var="$1"
+	local fileVar="${var}_FILE"
+	local def="${2:-}"
+	if [ "${!var:-}" ] && [ "${!fileVar:-}" ]; then
+		echo >&2 "error: both $var and $fileVar are set (but are exclusive)"
+		exit 1
+	fi
+	local val="$def"
+	if [ "${!var:-}" ]; then
+		val="${!var}"
+	elif [ "${!fileVar:-}" ]; then
+		val="$(< "${!fileVar}")"
+	fi
+	export "$var"="$val"
+	unset "$fileVar"
```

Comment:
Is `unset` needed here?

Reply:
I simply copied that function from https://github.com/docker-library/mysql/blob/b39f1e5e4ec82dc8039cecc91dbf34f6c9ae5fb0/8.0/docker-entrypoint.sh 
Its standard function used in many images :) - they are unset it because when you execute that function second time with same var it will crush
```bash
if [ "${!var:-}" ] && [ "${!fileVar:-}" ]; then
		echo >&2 "error: both $var and $fileVar are set (but are exclusive)"
		exit 1
fi
```